src/auth/local.strategy.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common'
+import { PassportStrategy } from '@nestjs/passport'
+import { AuthService } from '../auth/auth.service'
+import { Strategy } from 'passport-local'
+import { LoginUserDto } from './dto/login-user.dto'
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+  constructor(private readonly authService: AuthService) {
+    super({
+      usernameField: 'email',
+    })
+  }
+
+  async validate(data: LoginUserDto) {
+    const user = await this.authService.validateUser(data)
+    if (user instanceof UnauthorizedException) {
```

Comment:
this if condition will never be reached in case of an error since when a function throws an error it should stop the execution of this function.

This kind of implementation should used in a try-and-catch statement. In the catch block when checking for the type of the thrown exception

Reply:
Please remove this if condtion. the validateUser has already handled the errors for us